config-server.md

```diff
@@ -11,23 +11,36 @@ Proposal: Introduce optional config API in the Director to fetch property values
 
 ## API
 
-Following public API will be used by the Director to contact config server:
+The config-server API for create, read and delete is as follows:
 
-- GET /v1/config/&lt;some-key-path>
-  - whenever Director needs to retrieve a value it will use GET action
-  - {"path": "some-key-path", "value": "..."}
+- GET /v1/config/&lt;some-key-path>[?version=&lt;version>]
+  - returns the configuration value in the specified `version`. If none is specified, the latest version is returned.
+  - {"path": "some-key-path", "value": "...", "version": "..."}
 
 - PUT /v1/config/&lt;some-key-path>
-  - whenever Director generates a value it will be saved into the config server
-  - {"value": "..."}
+  - {"value": "...", "type": "..."}
+  - adds value as new latest version of the configuration value
+
+- DELETE /v1/config/&lt;some-key-path>[?retain-versions=[2,4..6]]
+  - if `retain-versions` is specified, all other versions are removed
 
 Values could be any valid JSON object.
 
 ### Authentication
 
 Director will use UAA for authentication with the config server. The Director will be configured with a UAA client and secret (similarly to how HM is configured).
 
-## Manifest configuration
+## CLI
+Go CLI should support the following commands:
+- `config login &lt;user> [&lt;password>]` - Get a UAA token for the user. If no &lt;password> is specified it is requested interactively.
```

Comment:
fwiw, when inside a Markdown code/pre block you can use `<` and `>` directly instead of `&lt;` and `>`. Makes it a bit more readable in plaintext form.